### DISCUSSION TITLE
start with Boolean command-line flags

### DISCUSSION THREAD

**Discussion by kalle07** [oobabooga/text-generation-webui#4618] [Q&A]
hey 

i installed the version where i start the 
start_windows.bat

so i add in line 
call python one_click.py %* --monkey-patch
also i can add --extensions xyz

it works but where i add 
rwkv_cuda_on 
sdp_attention 
and so one ? it dont work the same way
and "save ui defaults" dont work in my version 

thy for help

**Comment by HiroseKoichi** [comment]
For extensions, you add them after --extensions separated by spaces like this: --extensions suberboogav2 openai
For Boolean command-line flags, you add -- in front of them like this: --sdp_attention --rwkv_cuda_on

If you need more help figuring out what cli options you can use, then just run: python server.py --help

**Comment by kalle07** [comment]
so in addition if you have 
one_click.py 
search for(or similar) :
def launch_webui():
    run_cmd(f"python server.py --monkey-patch --...

**Reply by HiroseKoichi** [reply]
Yes, you could edit the file and add the flags directly to it, like this:
```
def launch_webui():
    run_cmd(f"python server.py --monkey-patch --sdp_attention --rwkv_cuda_on", environment=True)
```
But then you would have to open up the file each time you want to change the flags, and if you try to update via git pull, then it will prevent it because the file has been edited. What you should do instead is add them when you run one_click.py, like this: python one_click.py --monkey-patch --sdp_attention --rwkv_cuda_on

**Comment by DmitryVN** [comment]
Can I use this:
set COMMANDLINE_ARGS= --sdp_attention --rwkv_cuda_on
set CUDA_MODULE_LOADING=LAZY
set NUMEXPR_MAX_THREADS=24
in start_windows.bat?

**Reply by HiroseKoichi** [reply]
Like I said, it's best not to edit the "non-user" files so that when you run git pull to update, you don't get any errors. If you're wanting to just double-click a file, then you can create a separate .bat file on your desktop instead, like this:
```
@ECHO OFF

set COMMANDLINE_ARGS= --sdp_attention --rwkv_cuda_on
set CUDA_MODULE_LOADING=LAZY
set NUMEXPR_MAX_THREADS=24
start C:\PATH\TO\FOLDER\start_windows.bat
```
I don't use or own a Windows computer, so I can't really test if this works or not.

**Reply by DmitryVN** [reply]
It works, but Boolean command-line flags are not marked.
How do I check if these commands actually work?
Thanks

**Reply by HiroseKoichi** [reply]
Try putting the command-line arguments after the .bat file, like this:
```
@ECHO OFF

set CUDA_MODULE_LOADING=LAZY
set NUMEXPR_MAX_THREADS=24
start C:\PATH\TO\FOLDER\start_windows.bat --sdp_attention --rwkv_cuda_on
```
In order to easily see if they are working properly, you can add --verbose to the .bat file, generate something, and then check the terminal to see if the prompt is displayed.
